libraries/AP_BoardConfig/AP_BoardConfig_CAN.h

```diff
@@ -0,0 +1,121 @@
+#pragma once
+
+#include <AP_HAL/AP_HAL.h>
+#include "AP_BoardConfig.h"
+#include <AP_Common/AP_Common.h>
+#include <AP_Param/AP_Param.h>
+#include <sys/ioctl.h>
+
+#if HAL_WITH_UAVCAN
+#define UAVCAN_PROTOCOL_ENABLE  1
+#endif
+
+class AP_BoardConfig_CAN {
+public:
+    // constructor
+    AP_BoardConfig_CAN(void)
+    {
+        AP_Param::setup_object_defaults(this, var_info);
+    };
+
+    void init(void);
+
+    static const struct AP_Param::GroupInfo var_info[];
+
+#if HAL_WITH_UAVCAN
+    class CAN_var_info {
+        friend class AP_BoardConfig_CAN;
+
+    public:
+        CAN_var_info()
+        {
+            AP_Param::setup_object_defaults(this, var_info);
+        }
+
+        static const struct AP_Param::GroupInfo var_info[];
+
+    private:
+        AP_Int8 _can_enable;
+        AP_Int8 _can_debug;
+        AP_Int32 _can_bitrate;
+    };
+
+    class CAN_protocol_var_info {
+        friend class AP_BoardConfig_CAN;
+
+    public:
+        CAN_protocol_var_info() : _uavcan(nullptr)
+        {
+            AP_Param::setup_object_defaults(this, var_info);
+        }
+        static const struct AP_Param::GroupInfo var_info[];
+
+    private:
+        AP_Int8 _protocol_enable;
```

Comment:
Well, zero disables any high CAN protocol running in the driver, but it doesn't disable it, it will still be initialized. I think that just `protocol` or `can_protocol` would be better.